package.json

```
@@ -22,14 +22,18 @@
     "chroma-js"
   ],
   "sideEffects": false,
-  "main": "./dist/index.js",
-  "unpkg": "./dist/index.umd.js",
-  "module": "./dist/index.module.js",
-  "source": "./src/index.ts",
+  "main": "./dist/index.main.umd.js",
+  "module": "./dist/index.module.es.js",
   "types": "./dist/index.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/index.exports.import.es.mjs",
+      "require": "./dist/index.exports.require.cjs.js"
+    },
```

```suggestion
    ".": [
      {
        "import": "./dist/index.exports.import.es.mjs",
        "require": "./dist/index.exports.require.cjs.js",
        "default": "./dist/index.exports.require.cjs.js"
      },
      "./dist/index.exports.require.cjs.js"
    ],
```
otherwise it'll break in a few node 13 versions. you could also just omit "require" entirely, but you always need "default".

---

ah nice. i didn't know it supported an array. i'll give this a try this week

---

>  you could also just omit "require" entirely, but you always need "default".

I'm looking at `@rollup/plugin-node-resolve` and it says:

> By default, this plugin looks for the `['default', 'module', 'import']` conditions when resolving imports.
>
> [[source]](https://github.com/rollup/plugins/tree/9874740ca685b1600e4319365c21e8c1131e9d95/packages/node-resolve#exportconditions)

Do you know if the order matters there? I intentionally left out `default` there because I wanted the rollup plugin to resolve `module` before default

---

the order does matter, yes, so having "default" last is what you want.

Also, "module" isn't a condition, so it's super weird that that rollup would look for it.

---

I was looking deeper into this and the order of the conditions matched by `@rollup/plugin-node-resolve` does not seem to matter.

I was looking at a matching implementation [here](https://github.com/lukeed/resolve.exports/tree/6d29cfbabfea13a41cbb75b57bdf46f6b0bb5115#optionsconditions) with this nice explainer

> **Important:** The order specified within `options.conditions` does not matter.
> The matching order/priority is **always** determined by the `"exports"` map's key order.

…which made things click for myself.

This may have been what you meant but I was still confused 😅. Anyway I put the `default` back in there with the CJS bundle and removed `require`.

---

> Also, "module" isn't a condition, so it's super weird that that rollup would look for it.

This condition is noted in webpack's [package exports doc](https://webpack.js.org/guides/package-exports/#reference-syntax). It's nice if you wanted to target those bundlers but the output was redundant to the `import` entry so I removed it.